### PR TITLE
Add temperature to equations of state

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp [hunk 1]
```diff
@@ -133,6 +133,37 @@ class EquationOfState<IsRelativistic, 1> : public PUP::able {
...
+  }
+  virtual Scalar<DataVector> temperature_from_density(
+      const Scalar<DataVector>& rest_mass_density) const {
+    return Scalar<DataVector>{DataVector{get(rest_mass_density).size(), 0.0}};
+  }
+  /// @}
+
+  /// @{
+  /*!
+   * Computes the specific internal energy \f$\epsilon\f$ from the rest mass
```
Comment (kidder) [inline src/PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp]: RIP

File: src/PointwiseFunctions/Hydro/EquationsOfState/HybridEos.hpp [hunk 2]
```diff
@@ -32,14 +32,22 @@ namespace EquationsOfState {
...
  * where \f$p\f$ is the pressure, \f$\rho\f$ is the rest mass density,
  * \f$\epsilon\f$ is the specific internal energy, \f$p_{cold}\f$ and
  * \f$\epsilon_{cold}\f$ are the pressure and specific internal energy evaluated
  * using the cold EOS, and \f$\Gamma_{th}\f$ is the adiabatic index for the
  * thermal part.
+ *
+ * The temperature \f$T\f$ is defined as
+ *
+ * \f[
+ * T =
```
Comment (kidder) [inline src/PointwiseFunctions/Hydro/EquationsOfState/HybridEos.hpp]: RIP

**Review by kidder — CHANGES_REQUESTED** [review]

**Review by kidder — DISMISSED** [review]

**Comment by teukolsky** [comment]
Question: If p = rho epsilon (Gamma - 1), and p = rho T (ideal gas, p = n k T in usual units), then shouldn't we define T as (Gamma - 1) epsilon instead of just epsilon? Or are we supposed to remember to put the Gamma - 1 by hand afterward?

**Comment by nilsdeppe** [comment]
I think that's just an arbitrary choice. The temperature has to scale with the specific internal energy. I'm happy to change the definition to include the `(Gamma - 1)` if you prefer (I think that would be more consistent with SpEC anyway)

**Comment by teukolsky** [comment]
> 

I don't feel strongly about this - I was more concerned about documenting things more clearly. On balance, probably better to include the Gamma - 1. Then getting to Kelvins just involves fundamental constants, independent of Gamma.

**Comment by nilsdeppe** [comment]
I think that's a good argument for including the `Gamma - 1` (and it'll just be less confusing in general). I'll make that change when I rebase after the test fix for develop is merged!

**Comment by nilsdeppe** [comment]
Rebased and squashed in change of temperature definition for the ideal gas

**Review by kidder — DISMISSED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**